### ISSUE TITLE
Pylint 3.1.0 test failure with pytest 7

### ISSUE THREAD

**Issue by Thermi** [pylint-dev/pylint#9544]
### Bug description

```python
Pytest version 7.4.4, pipeline output here: https://gitlab.alpinelinux.org/Thermi/aports/-/jobs/1334679

Test failure during building:


=================================== FAILURES ===================================
____________________ test_functional[recursion_error_3159] _____________________
self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x7f5f8dc01a50>
    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong message(s) raised for "recursion_error_3159.py":
E       
E       Unexpected in testdata:
E         17: arguments-differ
pylint/testutils/lint_module_test.py:147: AssertionError
=========================== short test summary info ============================
FAILED tests/test_functional.py::test_functional[recursion_error_3159] - AssertionError: Wrong message(s) raised for "recursion_error_3159.py":
Unexpected in testdata:
  17: arguments-differ
====== 1 failed, 1825 passed, 327 skipped, 5 xfailed in 159.26s (0:02:39) ======
```
```


### Configuration

_No response_

### Command used

```shell
.testenv/bin/python3 -m pytest --benchmark-disable tests
```

### Pylint output

```python
=================================== FAILURES ===================================
____________________ test_functional[recursion_error_3159] _____________________
self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x7f5f8dc01a50>
    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong message(s) raised for "recursion_error_3159.py":
E       
E       Unexpected in testdata:
E         17: arguments-differ
pylint/testutils/lint_module_test.py:147: AssertionError
=========================== short test summary info ============================
FAILED tests/test_functional.py::test_functional[recursion_error_3159] - AssertionError: Wrong message(s) raised for "recursion_error_3159.py":
Unexpected in testdata:
  17: arguments-differ
====== 1 failed, 1825 passed, 327 skipped, 5 xfailed in 159.26s (0:02:39) ======
```

### Expected behavior

Test passes

### Pylint version

```shell
pytest 7.4.4
astroid 3.1.0
python 3.11.8
```

### OS / Environment

Alpine Linux Edge

**Comment by Pierre-Sassoulas** [comment]
Will be closed in https://github.com/pylint-dev/pylint/pull/9483

**Comment by Pierre-Sassoulas** [comment]
Closing as I cannot reproduce. I've seen that there's an issue but for pytest 8.0.0+ only (see https://github.com/pylint-dev/pylint/pull/9576#issuecomment-2088393182)
